src/org/rust/lang/inspections/ApproxConstantInspection.kt

```diff
@@ -0,0 +1,78 @@
+package org.rust.lang.inspections
+
+import com.intellij.codeInspection.LocalInspectionTool
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementVisitor
+import org.rust.lang.core.psi.RustLitExpr
+import org.rust.lang.core.psi.RustVisitor
+import java.util.*
+
+class ApproxConstantInspection : LocalInspectionTool() {
+
+    override fun getGroupDisplayName() = "Rust"
+    override fun getDisplayName() = "Approximate Constants"
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : RustVisitor() {
+            override fun visitLitExpr(o: RustLitExpr) {
+                analyzeLiteral(o.floatLiteral ?: return, holder);
+            }
+        }
+    }
+
+    private fun analyzeLiteral(psiElement: PsiElement, holder: ProblemsHolder) {
+        val constant = KNOWN_CONSTS
+            .find { isApproxConst(it.value, psiElement.text, it.minDigits) } ?: return
+
+        holder.registerProblem(psiElement,
+            "Approximate value of `f{32,64}::${constant.name}` found. Consider using it directly.");
+    }
+
+    /**
+     * Check if the given value is similar to the predefined constant
+     */
+    private fun isApproxConst(constant: Double, value: CharSequence, minDigits: Int): Boolean {
+        // check if the value has a certain minimum number of digits
+        if (value.length <= minDigits) {
+            return false
+        }
+
+        // round the constant to the same number of digits as the value and compare
+        val roundedConst = "%.${value.length - 2}f".format(Locale.ROOT, constant)
+        if (value == roundedConst) {
+            return true
+        }
+
+        // truncate the constant to the same number of digits as the value and compare
+        var truncatedConst = constant.toString();
+        if (truncatedConst.length > value.length) {
+            truncatedConst = truncatedConst.substring(0, value.length)
+        }
+
+        return value == truncatedConst
```

Comment:
I'm not sure that comparison of strings is a best approach. May be we should convert a constant to double and compare numerical values?